core/control_stream.go

```diff
@@ -28,72 +40,102 @@ type ServerControlStream interface {
 
 	// VerifyAuthentication verify the Authentication from client side.
 	VerifyAuthentication(verifyFunc func(auth.Object) (bool, error)) error
-	// AcceptStream accepts data stream from the request of client.
-	AcceptStream(context.Context) (DataStream, error)
+
+	// OpenStream reveives a HandshakeFrame from control stream and handle it in the function be passed in.
+	// if handler returns nil, There will return a DataStream and nil,
+	// if handler returns an error, There will return a nil and the error,
+	OpenStream(context.Context, func(*frame.HandshakeFrame) error) (DataStream, error)
 }
 
-// ClientControlStream defines the interface of client side control stream.
+// ClientControlStream is an interface that defines the methods for a client-side control stream.
 type ClientControlStream interface {
 	ControlStream
 
-	// Authenticate with credential, the credential will be sent to ServerControlStream to authenticate the client.
+	// Authenticate sends the provided credential to the server's control stream to authenticate the client.
 	Authenticate(*auth.Credential) error
-	// OpenStream request a ServerControlStream to create a new data stream.
-	OpenStream(context.Context, *frame.HandshakeFrame) (DataStream, error)
+
+	// RequestStream sends a HandshakeFrame to the server's control stream to request a new data stream.
+	// If the handshake is successful, a DataStream will be returned by the AcceptStream() method.
+	RequestStream(*frame.HandshakeFrame) error
```

Comment:
The flow for opening a stream is as follows:

1. The client calls RequestStream.
2. The server either opens a new stream or sends a HandshakeRejectedFrame to the client.
3. If the server opens a new stream, the client can accept the stream and return a data stream. If the server sends a HandshakeRejectedFrame, the client cannot return a data stream and must provide a reason for the rejection.